src/internal/operators/every.ts

```diff
@@ -1,9 +1,19 @@
 /** @prettier */
 import { Observable } from '../Observable';
-import { OperatorFunction } from '../types';
+import { OperatorFunction, TruthyTypesOf } from '../types';
 import { operate } from '../util/lift';
 import { OperatorSubscriber } from './OperatorSubscriber';
 
+export function every<T, S extends T>(
+  predicate: (value: T, index: number, source: Observable<T>) => value is S,
+  thisArg?: any
+): OperatorFunction<T, S>;
+export function every<T>(predicate: BooleanConstructor, thisArg?: any): OperatorFunction<T, TruthyTypesOf<T>>;
```

Comment:
LOL... I blame type system brain damage.

Reply:
Okay, now it will infer correctly using `Boolean` for `Observable<Falsy>` -> `Observable<false>`. Unfortunately, we cannot do the opposite. This is because of a few things:  1. `NaN` isn't typable, but it's "Falsy". and 2. TMK there's no way to create a "Truthy" type. The best you can do is verify that nothing appears to be Falsy, but that's no guarantee, because `0` is a `number` and only `0 as const` is `0` as far as the type system is concerned. 🤷